cdap-app-fabric/src/main/java/org/apache/twill/yarn/YarnTwillPreparer.java

```diff
@@ -385,46 +397,85 @@ private LocalFile createLocalFile(String name, Location location, boolean archiv
     return new DefaultLocalFile(name, location.toURI(), location.lastModified(), location.length(), archive, null);
   }
 
+  private void copyContents(Location from, Location to) throws IOException {
```

Comment:
You can use `Locations` `newInputSupplier` and `newOutputSupplier` together with `ByteStreams.copy`